icubam/backoffice/handlers/store.py

```diff
@@ -0,0 +1,43 @@
+import abc
+import tornado.escape
+
+from icubam.backoffice.handlers.base import BaseHandler
+
+"""A base handler for store-related operations (CRUD-like).
+Users must be authenticated to be able to call these operations. Instead of
+only checking the cookie, this class also checks if the cookie is valid, and
+corresponds to a real user in the database. If so, it forwards the user to
+the child classes, so derived classes know that they are dealing with a valid
+user of the system.
+"""
+class StoreHandler(BaseHandler, metaclass=abc.ABCMeta):
+
+  def get_logged_user(self):
+    userid = self.get_current_user()
+    if not userid:
+      return None
+
+    return self.store.get_user(int(tornado.escape.json_decode(userid)))
+
+  """Invokes the GET operation on the context of the user."""
+  def getForUser(self, user):
+    pass
+
+  """Invokes the POST operation on the context of the user."""
+  def postForUser(self, user):
+    pass
+
+  def validate_and_call(self, callback):
```

Comment:
i believe the `@tornado.web.authenticated` already checks if there is a current user, via cookie.
Then one way is to overload get_current_user() and add the user in like self._user ? Then you can simple add the decorator whereever you want authentication + access to the user...

Reply:
that's great. I didn't realize we can override get_current_user to return whatever we want :)
(too much C++ in the last days).